### PR TITLE
bpo-41137: Use utf-8 encoding while reading .pdbrc files

### PR DESCRIPTION
**Pull request by srinivasreddy**
https://bugs.python.org/issue41137

### CODE REVIEW

**Review by eamanu — APPROVED** [review]
good catch :)

**Comment by serhiy-storchaka** [comment]
Thank you for your PR @srinivasreddy, but it needs more changes:

1. In the pdb documentation specify explicitly the encoding of the `.pdbrc` file.
2. Add a note in What's New.
3. Update tests. It would be nice to have a test failing with non-UTF-8 encoding.

**Comment by srinivasreddy** [comment]
@serhiy-storchaka  Not sure why windows build is failing. Any idea why?

File: Lib/test/test_pdb.py [hunk 1]
```diff
@@ -1448,6 +1447,40 @@ def test_readrc_homedir(self):
...
+    def test_read_pdbrc_with_ascii_encoding(self):
+        script = textwrap.dedent("""
+            import pdb; pdb.Pdb().set_trace()
+            print('hello')
+        """)
+        save_home = os.environ.pop('HOME', None)
+        try:
+            with support.temp_cwd():
+                with open('.pdbrc', 'w') as f:
+                    f.write("Fran\u00E7ais\n")
```
Comment (serhiy-storchaka) [inline Lib/test/test_pdb.py]: It will not work on locales with encoding which does not allow to encode `\u00E7`. You need to find characters which are encodable with the locale encoding.

Also, it would be better to use more realistic example. Use a command which takes a non-ASCII argument and check that it works correctly.

File: Doc/library/pdb.rst [hunk 2]
```diff
@@ -241,10 +241,11 @@ middle of a quoted string.
    triple: debugger; configuration; file
 
 If a file :file:`.pdbrc` exists in the user's home directory or in the current
-directory, it is read in and executed as if it had been typed at the debugger
-prompt.  This is particularly useful for aliases.  If both files exist, the one
-in the home directory is read first and aliases defined there can be overridden
-by the local file.
+directory, it is read with ``'utf-8'`` encoding rather than system locale
```
Comment (serhiy-storchaka) [inline Doc/library/pdb.rst]: "rather than system locale encoding" is not needed here. It will rather confuse users. But please add the `versionchanged` directive in which describe the change.

**Review by serhiy-storchaka — COMMENTED** [review]
Please do not forget about What's New.

File: Doc/library/pdb.rst [hunk 3]
```diff
@@ -241,10 +241,11 @@ middle of a quoted string.
...
+encoding and executed as if it had been typed at the debugger prompt.This is
```
Comment (serhiy-storchaka) [inline Doc/library/pdb.rst]: And please minimize the diff to make reviewing easier. Do not rewrap the unchaged text. It is normal if some lines will be too short or slightly longer that PEP 8 recommendations.

File: Lib/test/test_pdb.py (hunk 1 above)
Comment (srinivasreddy) [reply Lib/test/test_pdb.py]: Now the build is passing. And I would like to keep this negative test case as it would, in utf-8 encoding gives "NameError: name 'Française' is not defined". Or do you still want me write the test you commented above?

File: Doc/library/pdb.rst (hunk 3 above)
Comment (srinivasreddy) [reply Doc/library/pdb.rst]: Current diff is the one the best I can come up. Not sure we would recommend a line with 120 chars?

File: Lib/test/test_pdb.py [hunk 4]
```diff
@@ -1448,6 +1447,40 @@ def test_readrc_homedir(self):
...
+            import pdb; pdb.Pdb().set_trace()
+            print('hello')
+        """)
+        save_home = os.environ.pop('HOME', None)
+        try:
+            with support.temp_cwd():
+                with open('.pdbrc', 'w', encoding='utf-8') as f:
+                    f.write("Fran\u00E7ais")
+
+                with open('main.py', 'w') as f:
```
Comment (methane) [inline Lib/test/test_pdb.py]:
```suggestion
                with open('main.py', 'w', encoding='utf-8') as f:
```

File: Lib/test/test_pdb.py [hunk 5]
```diff
@@ -1448,6 +1447,40 @@ def test_readrc_homedir(self):
                 if save_home is not None:
                     os.environ["HOME"] = save_home
 
+    def test_read_pdbrc_with_ascii_encoding(self):
```
Comment (methane) [inline Lib/test/test_pdb.py:1630]: I am not sure this test case is worth enough.

File: Doc/library/pdb.rst [hunk 6]
```diff
@@ -241,10 +241,14 @@ middle of a quoted string.
...
-directory, it is read in and executed as if it had been typed at the debugger
-prompt.  This is particularly useful for aliases.  If both files exist, the one
-in the home directory is read first and aliases defined there can be overridden
-by the local file.
+directory, it is read with ``'utf-8'`` encoding and executed as if it had been
+typed at the debugger prompt.  This is particularly useful for aliases.  If both
+files exist, the one in the home directory is read first and aliases defined there
+can be overridden by the local file.
+
+.. versionchanged:: 3.10
```
Comment (methane) [inline Doc/library/pdb.rst]:
```suggestion
.. versionchanged:: 3.11
```

File: Doc/whatsnew/3.10.rst [hunk 7]
```diff
@@ -82,6 +82,7 @@ New Features
 * :pep:`618`: The :func:`zip` function now has an optional ``strict`` flag, used
   to require that all the iterables have an equal length.
 
+* :file:`.pdbrc` is now read with ``'utf-8'`` encoding.
```
Comment (methane) [inline Doc/whatsnew/3.10.rst]: Move this line to 3.11.rst.

File: Lib/test/test_pdb.py [hunk 8]
```diff
@@ -1402,7 +1402,6 @@ def test_issue16180(self):
             'Fail to handle a syntax error in the debuggee.'
             .format(expected, stdout))
 
-
```
Comment (methane) [inline Lib/test/test_pdb.py]: revert unnecessary change.
